spec/adapter/base_spec.cr

```diff
@@ -1,32 +1,40 @@
 require "../spec_helper"
 
+Granite::Settings.adapters << Granite::Adapter::Pg.new({name: "FooBar123", url: "FAKE_DB"})
+
+class Foo < Granite::Base
+  adapter FooBar123
+end
+
 describe Granite::Adapter::Base do
-  describe "#env" do
-    context "no ${} exists" do
-      it "should return itself" do
-        url = "dummy://user:pswd@host:3333/database"
-        Granite::Adapter::Base.env(url).should eq url
+  describe "registration" do
+    it "should allow adapters to be registered" do
+      Granite::Settings.adapters.size.should eq 4
+
+      if mysql_adatper = Granite::Settings.adapters.find { |adapter| adapter.name == "mysql" }
+        mysql_adatper.url.should eq ENV["MYSQL_DATABASE_URL"]
+      end
+      if pg_adatper = Granite::Settings.adapters.find { |adapter| adapter.name == "pg" }
+        pg_adatper.url.should eq ENV["PG_DATABASE_URL"]
+      end
+      if sqlite_adapter = Granite::Settings.adapters.find { |adapter| adapter.name == "sqlite" }
+        sqlite_adapter.url.should eq ENV["SQLITE_DATABASE_URL"]
+      end
+      if foo_adapter = Granite::Settings.adapters.find { |adapter| adapter.name == "FooBar123" }
+        foo_adapter.url.should eq "FAKE_DB"
```

Comment:
these tests can fail with a false positive if the find returns nil. Is there an easy way to write them so that they'll throw when find returns nil?

Reply:
Could just do the find off adapters array and `not_nil!` it?